packages/enclave-contracts/contracts/Enclave.sol

```diff
@@ -126,12 +260,12 @@ contract Enclave is IEnclave, OwnableUpgradeable {
     //                                                        //
     ////////////////////////////////////////////////////////////
 
+    /// @inheritdoc IEnclave
     function request(
         E3RequestParams calldata requestParams
-    ) external payable returns (uint256 e3Id, E3 memory e3) {
-        // TODO: allow for other payment methods or only native tokens?
-        // TODO: should payment checks be somewhere else? Perhaps in the E3 Program or ciphernode registry?
-        require(msg.value > 0, PaymentRequired(msg.value));
+    ) external returns (uint256 e3Id, E3 memory e3) {
+        uint256 e3Fee = getE3Quote(requestParams);
+        require(e3Fee > 0, PaymentRequired(e3Fee));
```

Comment:
is this to prevent some sort of param combination that might cause getE3Quote to to return zero? perhaps it could be moved inside getE3Quote directly?